openssl/aes.go

```diff
@@ -116,32 +107,19 @@ func (c *aesCipher) Decrypt(dst, src []byte) {
 		panic("crypto/aes: output not full block")
 	}
 	if c.dec_ctx == nil {
-		c.dec_ctx = C._goboringcrypto_EVP_CIPHER_CTX_new()
-		if c.dec_ctx == nil {
-			panic("cipher: unable to create EVP cipher ctx")
-		}
-
-		k := (*C.uchar)(unsafe.Pointer(&c.key[0]))
-
-		if C.int(1) != C._goboringcrypto_EVP_CipherInit_ex(c.dec_ctx, c.cipher, nil, k, nil, C.GO_AES_DECRYPT) {
-			panic("cipher: unable to initialize EVP cipher ctx")
+		var err error
+		c.dec_ctx, err = newCipherCtx(c.cipher, C.GO_AES_DECRYPT, c.key, nil)
+		if err != nil {
+			panic(err)
 		}
 	}
-	// Workaround - padding detection is broken but we don't need it
-	// since we check for full blocks
-	if C._goboringcrypto_EVP_CIPHER_CTX_set_padding(c.dec_ctx, 0) != 1 {
-		panic("crypto/cipher: could not disable cipher padding")
-	}
-	outlen := C.int(0)
-	C._goboringcrypto_EVP_CipherUpdate(c.dec_ctx, (*C.uchar)(unsafe.Pointer(&dst[0])), &outlen, (*C.uchar)(unsafe.Pointer(&src[0])), C.int(aesBlockSize))
+
+	C.go_openssl_EVP_DecryptUpdate_wrapper(c.dec_ctx, base(dst), base(src), aesBlockSize)
```

Comment:
FYI: all `C.go_openssl_EVP_*_wrapper` functions are there just to avoid allocating `outlen` just to discard it without being used.